test cases/unit/106 underspecified mtest/meson.build

```diff
@@ -0,0 +1,8 @@
+project('underspecified deps', 'c')
+
+runner = find_program('runner.py')
+exe1 = executable('main1', 'main.c')
+exe2 = executable('main2', 'main.c')
```

Comment:
For completeness one of these two should have `build_by_default: false`.

Reply:
That's not really what I'm testing here, I'm testing that `meson test` needs to build "all". I'm not testing what is included in "all".

Reply:
But it is relevant for this. Rebuilding needs to work even if the end user has tagged a test executable as `build_by_default: false` so we should test it.